cactus/plugin/builtin/pagecontext.py

```diff
@@ -0,0 +1,120 @@
+#coding:utf-8
+import logging
+import re
+
+
+logger = logging.getLogger(__name__)
+
+
+class PageContextPlugin(object):
+    """
+    A plugin to extract context variables from the top of the file
+
+    TODO:
+    - read style of metadata from config
+    - read splitChar from config
+    """
+
+    def preBuildPage(self, page, context, data):
+        """
+        Update the page context with the context variables from the file
+        and delete them
+        """
+        if not page.is_html() or not data:
+            return context, data
+
+        lines = data.splitlines()
+
+        # Look into the file to decide on the style of metadata
+        # TODO: make that configurable
+        if lines[0].strip() == u'---':
```

Comment:
Adding the following to the test cases:

            ([], {}, [])

Results in:

    ======================================================================
    ERROR: test_multimarkdown_style (cactus.tests.test_plugins.TestPageContextPlugin)
    ----------------------------------------------------------------------
    Traceback (most recent call last):
      File "/Users/thomas/dev/cactus/cactus/tests/test_plugins.py", line 230, in test_multimarkdown_style
        plugin.multimarkdown_style(lines),
      File "/Users/thomas/dev/cactus/cactus/plugin/builtin/page_context.py", line 88, in multimarkdown_style
        if splitChar not in lines[0] and lines[0].strip() != u'---':
    IndexError: list index out of range
    
    ======================================================================
    ERROR: test_simple_style (cactus.tests.test_plugins.TestPageContextPlugin)
    ----------------------------------------------------------------------
    Traceback (most recent call last):
      File "/Users/thomas/dev/cactus/cactus/tests/test_plugins.py", line 162, in test_simple_style
        plugin.simple_style(lines),
      File "/Users/thomas/dev/cactus/cactus/plugin/builtin/page_context.py", line 77, in simple_style
        return context, lines[i:]
    UnboundLocalError: local variable 'i' referenced before assignment


(Note: I realize some of that broken code was actually present before)